### PR TITLE
for wizard Step 2 create mixins for sideloading of has_many / belongs_to resources in API calls

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/controllers/fusor/api/v21/deployments_controller.rb [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
+                                        :update, :destroy,
+                                        :check_mount_point,
+                                        :deploy,
+                                        :redeploy,
+                                        :validate,
+                                        :log,
+                                        :openshift_disk_space,
+                                        :compatible_cpu_families]
+
+  include Fusor::Api::V21::DiscoveredHostsMixin
```
Comment (cfchase) [inline app/controllers/fusor/api/v21/deployments_controller.rb:31]: Let's not mix in any external API calls into the deployment object and side load them.  That would be painfully slow to load data.

Instead, we need to refactor the client to only load those when necessary.  That's probably on the initial load of the application in the background without holding up the route load, or on a page where we display the data.  The deployment object can just keep it's IDs and we can set them, but we should keep the API data loads to an absolute minimum.

The only other choice would be to cache the data in our database, but that gets complex as to when to refresh that data when satellite gets modified without Fusor's knowledge.

Comment (isratrade) [reply app/controllers/fusor/api/v21/deployments_controller.rb:31]: @cfchase, I agree that it's better performance to call API's when needed, and not side load, but as a first sprint iteration, I wanted to keep the front-end changes to a minimum which means fusor-server generating the same API responses as fusor/server plugin + foretello_api_v21 plugin.

Comment (cfchase) [reply app/controllers/fusor/api/v21/deployments_controller.rb:31]: @isratrade OK, I'll treat this as a "todo" then and we'll handle it at the same time the front end is being updated.

**Comment by cfchase** [comment]
ACK

**Pull request merged**